watertrading/code/MainScript.py

```diff
@@ -0,0 +1,53 @@
+##################################################
+# PYTHON IMPORTS
+import json
+import pandas as pd
+from datetime import datetime, timedelta
+from pyomo.opt import TerminationCondition
+import pyomo.environ
+from pyomo.environ import value
+from pyomo.opt import SolverFactory
+import BuildModel
+from os import getcwd
+
+
+##################################################
+# RUN OPTIMIZATION ALGORITHM
+def run_optimization_models(REQUESTS,DISTANCE,MATCHES,PROFITS,Update_distance_matrix=True,save_dir=getcwd(),verbose=True,OnlineMode=False,filter_ON=False,filter_date=None):
+
+    df_producer, df_consumer, df_midstream, df_restrictions, df_distance, df_time = BuildModel.get_data(REQUESTS,DISTANCE,Update_distance_matrix,OnlineMode=OnlineMode,filter_ON=filter_ON,filter_date=filter_date)
+
+    # Add some output and formatting
+    spacer="#"*50
+
+    # Run first optimization model
+    output_s = f"""
+    {spacer}
+    # Starting Optimization Run: Clearing Formulation
+    {spacer}
+    """
+    if verbose: print(output_s)
+    
+    water_sharing = BuildModel.create_model(df_restrictions, df_producer, df_consumer, df_midstream, df_distance, df_time,default={})
+    #solver = SolverFactory('scip', solver_io='nl')
+    #solver = SolverFactory("gurobi")
+    solver = SolverFactory("glpk")
```

Comment:
Why is GLPK your solver of choice here? I have been cautioned against using GLPK by other PSERT members.

Reply:
My Gurobi license was up, and I needed a quick fix.